WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -167,23 +167,24 @@ final class EditableOrderViewModel: ObservableObject {
 
     /// View model for the product list
     ///
-    lazy var addProductViewModel = {
+    var handleProductsViewModel: ProductSelectorViewModel {
```

Comment:
In the [OrderForm](https://github.com/woocommerce/woocommerce-ios/blob/0236d06f37493a0b4ca75719eeec7f0a74b9248d/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/OrderForm.swift#L285) class it expects it to be lazy:

`viewModel.addProductViewModel.clearSearchAndFilters()`

Did you take that into account?

Furthermore, with it being a property, I would expect it as well that it was created once, without being generated every time. If we want to do that, I would create a function to make it clear:

`func generateProductSelectorViewModel()`

From [here](https://stackoverflow.com/questions/24035276/computed-read-only-property-vs-function-in-swift):

> It seems to me that it's mostly a matter of style: I strongly prefer using properties for just that: properties; meaning simple values that you can get and/or set. I use functions (or methods) when actual work is being done

Reply:
I believe I need some guidance on how to move forward with this bit:

For creating the `ProductSelectorViewModel` I experimented with 3 options: 
1. Lazy ( currently in trunk)
2. Computed (changes in this PR)
3. Method call

And only the 2nd option ( computed ) seems to work as we want. The problem is within the `Flow 2` explained above in the `Testing Instructions`:
- Go to Orders > `+` > `+ Add Product`
- Select a Product > Tap on Done > See how is added to the Order
- Tap on the `-` (minus) icon to remove the product from the Order > Confirm by tapping `Remove Product from Order`
- Tap on + Add Product > See how the Product toggle appears as unselected

Only making the view model computed we seem to update the selected products properly, the other two will keep the state and products will appear selected despite had been removed from the Order.

From what I understand, if the view model being lazy we're initializing this once with whatever values we have on `selectedItemIDs`, so values are not updated as needed.

The same seems to happen with the 3rd option by using a function call, which was unexpected as I would expect that this is re-instantiated with the new values. Could have something to do with the view model being an `@ObservableObject`? I attempted to add a `deinit` breakpoint to see when it's reinstantiated but this doesn't seem to be triggered, so that could be why both non-computed approaches show the same issue.

Perhaps there's a way to not compute the whole view model, but just what's needed?

Reply:
> Method call

I found the way to make the method call approach work by also changing the [ProductSelectorView.Configuration](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/OrderForm.swift#L354) initializer into an static method, so we keep the model updated when the view calls it. However, this causes other UI issues that need to be resolved, so I'm switching the PR back to `draft` for the moment.